### ISSUE TITLE
Error 500 when trying to unblock IP or remove IP from exception list

### ISSUE THREAD

**Issue by Jumanji1010** [codeling/bfstop#191]
I tried to unblock a blocked IP which gave the 500 error.
Same happened when trying to delete an IP from the Exception list.

Adding to the exception or the blocked list does work.

**Comment by codeling** [comment]
What do you mean with "Exception list" - maybe the "Allow list" (/ "White list " as it was called in previous versions)?
Which version of bfstop does this happen with, which Joomla version?
Do you get any more specific error message or entries in the log file? If not, can you try turning up the logging on your Joomla installation to get more information?

**Comment by codeling** [comment]
> see below:

I see no error message below unfortunately :(

**Comment by Jumanji1010** [comment]
I attached two screen shots  but they seem to got lost. Here the text from one of the screenshots, hope that helps or let me please know, how I can sent you the pictures…

FatalError
HTTP 500 Whoops, looks like something went wrong.
Compile Error: Declaration of BFStopController AllowList: :getModel ($name = 'allowlist', $prefix = "bfstopmode..
must be compatible with Joomla\CMS\VC\Controller\BaseController:getModel($name = ", $prefix =
. Sconfig
=1)
Exception!
Exception
Stack Trace
Symfony\Component\ErrorHandler\Error)
FatalError
in/homepages/28/d146014317/htdocs/entwicklung/administrator/components/com_bfstop/controllers/allowlist.php(line15)
10
jimport('joomla.application.component.controlleradmin');
12
13.
14.
15.
16
17.
18
class BFStopControllerAllowList extends ControllerAdmin
public function getModel($name
= 'allowlist', $prefix
=
'bfstopmodel')
$model = parent : :getModel($name, $prefix, array( 'ignore_request'
return $model;
=>
true)):


Mit freundlichen Grüßen

**Comment by chdh** [comment]
I have the same error with BFStopControllerBlockList and used the following patch to solve it:

old: `public function getModel($name = 'blocklist', $prefix = 'bfstopmodel')`
new: `public function getModel($name = 'blocklist', $prefix = 'bfstopmodel', $config = [])`

Error message in my case:

```
Fatal error: Declaration of BFStopControllerBlockList::getModel($name = 'blocklist', $prefix = 'bfstopmodel')
must be compatible with Joomla\CMS\MVC\Controller\BaseController::getModel($name = '', $prefix = '', $config = [])
in .../administrator/components/com_bfstop/controllers/blocklist.php on line 14
```

The same problem exists in:
administrator/components/com_bfstop/controllers/allowlist.php
administrator/components/com_bfstop/controllers/blocklist.ph
administrator/components/com_bfstop/controllers/htblocklist.php
administrator/components/com_bfstop/controllers/whitelist.php

It's probably a consequence of switching to PHP 8.

**Comment by codeling** [comment]
Thanks for the reminder!
`administrator/components/com_bfstop/controllers/whitelist.php` is actually a remainder of a previous version of bfstop.

For the rest, this is now fixed [with this commit](https://github.com/codeling/com_bfstop/commit/d372e616a95fab99b91f4d6df02045d238b93c36). The next release (probably 1.5.0, to be released some time in January) will include this fix!

**Comment by codeling** [comment]
Finally got a release out that fixes this issue [1.4.6](https://github.com/codeling/bfstop/releases/tag/1.4.6). This release should also get distributed already through Joomla's update system.
Let me know if you still encounter any issues with that version!
